### PR TITLE
Fix Dockerfiles to create valid openSUSE images and update docker/README.md

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
Fix dockerfiles and make the openSUSE images available from docker hub.

This PR depends on [that PR](https://github.com/os-autoinst/openQA/pull/1236) and [that other PR](https://github.com/os-autoinst/openQA/pull/1237).

Made during Hackweek 15 for project [openQA-butler](https://hackweek.suse.com/15/projects/openqa-butler)

### CODE REVIEW

**Comment by okurz** [comment]
* better rebase and get rid of the merge commits
* better subject line for PR

File: docker/README.md [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
-You can either build the images locally, or get our "latest stable" version from the Docker hub. We recommend using the Docker hub option.
+You can either build the images locally, or get the images from the Docker hub. We recommend using the Docker hub option.
 
 ## Download images from the Docker hub
 
+**openSUSE images**
+
+    docker pull slindomansilla/openqa_data
+    docker pull slindomansilla/openqa_webui
+    docker pull slindomansilla/openqa_worker_default
```
Comment (okurz) [inline docker/README.md]: can we find an "opensuse" project on the docker hub? There might be one already. If not, at least use "suse"?

File: docker/README.md [hunk 2]
```diff
@@ -1,18 +1,28 @@
 # Get docker images
 
-You can either build the images locally, or get our "latest stable" version from the Docker hub. We recommend using the Docker hub option.
+You can either build the images locally, or get the images from the Docker hub. We recommend using the Docker hub option.
 
 ## Download images from the Docker hub
 
+**openSUSE images**
```
Comment (okurz) [inline docker/README.md]: why not use proper header instead of marking as bold?

File: docker/README.md [hunk 3]
```diff
@@ -28,7 +38,8 @@ It's also possible to use `tests` and `factory` from within the `openqa_data` co
 to ditch `openqa_data` container altogether (so you have only `webui` and `worker` containers and data is loaded and saved completely into your host
 system). If this is what you want, refer to [Keeping all data in the Data Volume Container] and [Keeping all data on the host system] sections respectively.
 
-## Update firewall rules
+**Update firewall rules (Fedora)**
```
Comment (okurz) [inline docker/README.md]: -1. should not replace headers by just bold printing. This **is** a header

File: docker/README.md [hunk 4]
```diff
@@ -37,18 +48,27 @@ containers to communicate with each other - it prevents worker to connect to Web
...
-## Create directory structure
+**Create directory structure**
+
+In case you want to have the big files (isos and disk images, test and needles) outside of the docker volume container,
+you should create this file structrure from within the directory you are going to execute the container.
 
     mkdir -p data/factory/{iso,hdd} data/tests
 
-It is also necessary to either run all containers in privileged mode, or set selinux properly:
+It could be necessary to either run all containers in privileged mode, or set selinux properly. If you are having problems with it, run this command:
```
Comment (okurz) [inline docker/README.md]: under which circumstances is it not necessary?

File: docker/README.md [hunk 5]
```diff
@@ -37,18 +48,27 @@ containers to communicate with each other - it prevents worker to connect to Web
...
 
     chcon -Rt svirt_sandbox_file_t data
 
 ## Run the Data & WebUI containers
 
+**All data within the volume data container**
+
+    # openSUSE
+    docker run -d -h openqa_data --name openqa_data -v `pwd`/data/factory:/data/factory -v `pwd`/data/tests:/data/tests slindomansilla/openqa_data
+    docker run -d -h openqa_webui --name openqa_webui --volumes-from openqa_data -p 80:80 -p 443:443 slindomansilla/openqa_webui
```
Comment (okurz) [inline docker/README.md]: now you are replicating the block just because of the name "fedoraqa"? there should be less duplication. There might be the possibility to use the same docker hub project/user

File: docker/README.md [hunk 6]
```diff
@@ -57,27 +77,33 @@ create the API keys in the OpenQA's web interface, and store the configuration i
 
 ### Change the OpenID provider
 
-https://id.fedoraproject.org/ is set as a default OpenID provider. To change it, run:
+https://www.opensuse.org/openid/user/ is set as a default OpenID provider. To change it, run:
 
-    docker exec -ti openqa_data /scripts/set_openid
+    docker exec -it openqa_data /scripts/set_openid
```
Comment (okurz) [inline docker/README.md]: why?

File: docker/README.md (hunk 4 above)
Comment (SergioAtSUSE) [inline docker/README.md]: Actually the question is under which circumstances is it necessary. I have never needed to execute this command to make it work. And I only need to execute the worker container in privileged mode, so the virtual machine inside the container can work properly. I would remove this paragraph, but I preferred to maintain this information as hint in case someone eventually get this error. So I changed the "must do" speech to "do it if you get the error" speech.

File: docker/README.md (hunk 5 above)
Comment (foursixnine) [inline docker/README.md]: Using fedora as a name for something related to opensuse, calls for ambiguity, i would replace in both blocks:

 `slindomansilla/openqa_data` and `fedoraqa/openqa_data` by "image/provider" and clarify that "image/provider" could be either or.

File: docker/README.md (hunk 2 above)
Comment (SergioAtSUSE) [inline docker/README.md]: It is not a header, only a marker. Making is bold is only a way to find it faster with the eyes.

File: docker/README.md (hunk 3 above)
Comment (SergioAtSUSE) [inline docker/README.md]: Since we are describing Fedora and openSUSE images, it is no more a header, because there are text after that which is not inside this "section of text". It become an optional action to do if you are using Fedora images and then it is only a marker. Just maintained bold for eye-search.

But I can make it an own section and the text after that too if you prefer.

File: docker/README.md (hunk 1 above)
Comment (SergioAtSUSE) [inline docker/README.md]: Waiting for feedback of the maintainer of the account [docker hub opensuse](https://hub.docker.com/u/opensuse/)

File: docker/README.md (hunk 5 above)
Comment (SergioAtSUSE) [inline docker/README.md]: No, these are completely different images. The images under fedoraqa are using as operating system Fedora. the ones under slindomansilla are using openSUSE.

It could be, that an organisation openqa is created on docker hub, and then we could save the images on repositories opensuse and fedora. But I am not the maintainer of the fedora images. I don't have the dockerfile to build them and the ones under fedoraqa don't work.

Having the opensuse images under fedoraqa (domain for which I don't have push permission) would also not be good organized.

It is like the documentation for openQA when installing on openSUSE Leap, openSUSE Tumbleweed or Fedora. You have to duplicate the line "zypper in..." if you want to include all the flavors.

File: docker/README.md (hunk 6 above)
Comment (SergioAtSUSE) [inline docker/README.md]: That is only a personal choice.
One reason is unification: in this readme file, it sometimes appeared as -it and sometimes as -ti. And I would like to make it all the same order.
The other reason is that in the docker documentation it always appear as -it. So I chose this one.

Should I revert this change?

Comment (okurz) [inline docker/README.md]: no, it's fine

File: docker/README.md (hunk 2 above)
Comment (aaannz) [inline docker/README.md]: Any reason why not to make it a 3rd level header? Later in the docs there is a 3rd level header like this just for one sentence, IMO I would appreciate a link from index directly to download commands.

File: docker/README.md (hunk 1 above)
Comment (aaannz) [inline docker/README.md]: Please, let us know when you get the response.

**Comment by aaannz** [comment]
LGTM, I would like to wait for the response if this could be included at opensuse docker hub.

File: docker/README.md (hunk 2 above)
Comment (SergioAtSUSE) [inline docker/README.md]: Hello @aaannz,

It just didn't look as a header to me. But I have no problem to change it. Done.

But I don't understand what you mean by having a link from index to download commands.
Do you mean I should create an index in this README.md file? or in [that one](https://github.com/os-autoinst/openQA/blob/master/README.asciidoc)

Regards.

Comment (aaannz) [inline docker/README.md]: Yeah, I meant for table of contents, but there for markdown AFAIK there is no automatic way to do it (similar to :toc: as in asciidoc). So don't bother with it now. When it will be available, or this readme is converted to asciidoc, there will be one. I don't favour manually created ToCs.

And in the meantime, github generates direct links for markdown headers.

**Comment by SergioAtSUSE** [comment]
As the images are blocking this PR, I remove them from the documentation.
So this PR can be merged without this conflict.

**Comment by aaannz** [comment]
@okurz Anything more from your side?

**Comment by okurz** [comment]
LGTM
